{
  OperationContext operationContext=getInstrumentedContext();
  try {
    blob.delete(operationContext,lease);
  }
 catch (  StorageException e) {
    LOG.error("Encountered Storage Exception for delete on Blob: {}, Exception Details: {} Error Code: {}",blob.getUri(),e.getMessage(),e.getErrorCode());
    if (e.getErrorCode() != null && e.getErrorCode().equals("BlobNotFound") && operationContext.getRequestResults().size() > 1 && operationContext.getRequestResults().get(0).getException() != null) {
      LOG.debug("Swallowing delete exception on retry: {}",e.getMessage());
      return;
    }
 else {
      throw e;
    }
  }
 finally {
    if (lease != null) {
      lease.free();
    }
  }
}
